src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_ForEachStatement.cs

```diff
@@ -125,6 +125,14 @@ private BoundStatement RewriteEnumeratorForEachStatement(BoundForEachStatement n
             //     V v = (V)(T)e.Current;
             //     /* node.Body */
             // }
+
+            // The scope of the iteration variable is the foreach statement syntax.
```

Comment:
And the answer is - we create 2 closures (display classes). So the change is correct - we need to associate this block with a distinct syntax from the body of the statement (which represents a different closure). Just the comment is wrong.